lib/openid_connect/document.ex

```diff
@@ -20,6 +20,8 @@ defmodule OpenIDConnect.Document do
                           60 * 60
                         )
 
+  @max_document_size 1024*1024*1024
```

Comment:
I'm curious what the reason for the limit is?

Reply:
@bcardarella we allow a user to configure their own OIDC providers, you can make an endpoint with an extremely large OIDC document and we will try to cache it in a GenServer. This is probably not ideal (and should not be a part of this PR) and should rather stream-read the response and stop after a certain threshold, but better than nothing for time being ©.

Reply:
@bcardarella I've already pushed a fix that reads it in a stream, allows configuring max size, and opened a different PR that doesn't include changes not related to this PR :).